provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/task/RegistrationResult.java

```diff
@@ -27,4 +28,34 @@
         this.payload = jsonPayload;
         this.provisioningDeviceClientStatus = dpsStatus;
     }
+
+    void setRegistrationId(String registrationId)
+    {
+        this.registrationId = registrationId;
+    }
+
+    void setCreatedDateTimeUtc(String createdDateTimeUtc)
+    {
+        this.createdDateTimeUtc = createdDateTimeUtc;
+    }
+
+    void setStatus(String status)
+    {
+        this.status = status;
+    }
+
+    void setSubstatus(ProvisioningDeviceClientSubstatus substatus)
+    {
+        this.substatus = substatus;
+    }
+
+    void setETag(String eTag)
+    {
+        this.eTag = eTag;
+    }
+
+    void setLastUpdatesDateTimeUtc(String lastUpdatesDateTimeUtc)
+    {
+        this.lastUpdatesDateTimeUtc = lastUpdatesDateTimeUtc;
+    }
```

Comment:
Any reason not to use Lombok? Might save us some code.

Reply:
The reason I'm not using lombok here is because the fields themselves are defined in the super class. I'd add some protected or package private lombok setters in the super class instead, but the setters are only used within this class's package, and it wouldn't make sense to publicly expose setters for this read-only class